### ISSUE TITLE
TechDocs repo_url config breaks page render

### ISSUE THREAD

**Issue by Andrew-Wichmann** [backstage/backstage#5670]
## Expected Behavior

The page should render if a gitlab repo_url is provided.

## Current Behavior

The page crashes on load with the message:

```
react_devtools_backend.js:2557 Error: Missing required config value at 'integrations.github'
    at M.getConfigArray (index.esm.js:104)
    at index.esm.js:385
    at index.esm.js:1118
    at Array.forEach (<anonymous>)
    at K (index.esm.js:1117)
    at index.esm.js:1232
    at ro (react-dom.production.min.js:211)
    at gu (react-dom.production.min.js:257)
    at e.unstable_runWithPriority (scheduler.production.min.js:19)
    at Wl (react-dom.production.min.js:122)
 ```

## Steps to Reproduce

1. Setup Backstage without a github integration
2. Use mkdocs to create a techdocs site with a repo_url set.

## Context

We were able to use use techdocs before adding the repo_url, but we wanted to add the repo_url to get the "edit button" feature for faster editing.

We don't use GitHub at all, so we don't have a configuration.

## Your Environment

mkdocs.yml
```
site_name: "Mintel Event Bus"

site_url: https://backstage.int.everest-prod.aws.mintel.com/docs/default/System/event-bus

repo_url: https://gitlab.com/mintel/everest/event-bus/events/mintel-events-protobuf

nav:
  - Home: README.md

plugins:
  - techdocs-core
```

app-config.yaml
```
app:
  title: Mintel Backstage
  baseUrl: http://localhost:3000

organization:
  name: Mintel

backend:
  baseUrl: http://localhost:7000
  listen:
    port: 7000
  csp:
    connect-src: ["'self'", 'http:', 'https:']
  cors:
    origin: http://localhost:3000
    methods: [GET, POST, PUT, DELETE]
    credentials: true
  # config options: https://node-postgres.com/api/client
  database:
    client: pg
    connection:
      host: ${PG_HOST}
      database: ${PG_DATABASE}
      port: ${PG_PORT}
      user: ${PG_USER}
      password: ${PG_PASSWORD}
      # https://node-postgres.com/features/ssl
      #ssl: require # see https://www.postgresql.org/docs/current/libpq-ssl.html Table 33.1. SSL Mode Descriptions (e.g. require)
        #ca: # if you have a CA file and want to verify it you can uncomment this section
        #  $file: <file-path>/ca/server.crt
  # workingDirectory: /tmp # Use this to configure a working directory for the scaffolder, defaults to the OS temp-dir

integrations:
  gitlab:
    - host: gitlab.com
      token: ${GITLAB_TOKEN}

# proxy:
#   '/test':
#     target: 'https://example.com'
#     changeOrigin: true

techdocs:
  builder: 'external'
  publisher:
    type: 'awsS3'
    awsS3:
      bucketName: ${DOCS_S3_BUCKET}

auth:
  environment: ${STAGE}
  providers:
    gitlab:
      prod:
        clientId: ${GITLAB_APP_ID}
        clientSecret: ${GITLAB_APP_SECRET}
      development:
        clientId:<redacted>
        clientSecret: :<redacted>

catalog:
  rules:
    - allow: [Component, Resource, System, API, Group, User, Domain, Template, Location]
  locations:
    - type: url
      target: https://gitlab.com/mintel/everest/mintel-backstage-org-config/-/blob/master/catalog-info.yaml
```

**Comment by adamdmharvey** [comment]
I tried to take a quick peek at this, and it would seem to be something related to the ScmIntegrations in the `@backstage/integrations` package I think, since the error itself isn't from TechDocs purely, but TechDocs calling into the ScmIntegration and throwing that error you show.

@Andrew-Wichmann , can you specify what release of Backstage or what versions of your `@backstage` core packages you're on?

Also, if you don't mind my asking, is your **repository** URL truly: https://gitlab.com/mintel/everest/event-bus/events/mintel-events-protobuf ?

Based on your later example of a specific entity file, I feel like you may have an extra path in there?  i.e., is Mintel your account, everest your group, and event-bus your repo?  if so then what is `events/nintel-events-protobuf` if not a folder path INSIDE of your repo?  I do think the code is expecting just a repo name there, and so maybe it's getting confused somehow? But I'll admit to not knowing the specifics of how GitLab formulates URLs (nor do I have access to a specific instance or account). So just throwing questions out a bit in the dark. :)

e.g., comparing these...
```
https://gitlab.com/mintel/everest/event-bus/events/mintel-events-protobuf
https://gitlab.com/mintel/everest/mintel-backstage-org-config/-/blob/master/catalog-info.yaml
```

**Comment by Fox32** [comment]
I wonder which version of Backstage you are using. There was a recent fix for this in #5461.

**Comment by Andrew-Wichmann** [comment]
> Also, if you don't mind my asking, is your repository URL truly: https://gitlab.com/mintel/everest/event-bus/events/mintel-events-protobuf ?

No, that's truly the path `mintel/everest/event-bus/events/` are all groups in gitlab `mintel-events-protobuf` is the repo.

> I wonder which version of Backstage you are using. There was a recent fix for this in #5461.

```
@backstage/backend-common@0.6.3
@backstage/catalog-client@0.3.10
@backstage/catalog-model@0.7.7
@backstage/cli-common@0.1.1
@backstage/cli@0.6.9
@backstage/config-loader@0.6.0
@backstage/config@0.1.4
@backstage/core-api@0.2.17
@backstage/core@0.7.6
@backstage/dev-utils@0.1.15
@backstage/errors@0.1.1
@backstage/integration-react@0.1.1
@backstage/integration@0.5.1
@backstage/plugin-api-docs@0.4.11
@backstage/plugin-app-backend@0.3.11
@backstage/plugin-auth-backend@0.3.8
@backstage/plugin-catalog-backend@0.8.0
@backstage/plugin-catalog-import@0.5.3
@backstage/plugin-catalog-react@0.1.4
@backstage/plugin-catalog@0.5.5
@backstage/plugin-org@0.3.12
@backstage/plugin-proxy-backend@0.2.6
@backstage/plugin-scaffolder-backend@0.10.0
@backstage/plugin-scaffolder@0.9.1
@backstage/plugin-search-backend-node@0.1.3
@backstage/plugin-search@0.3.4
@backstage/plugin-tech-radar@0.3.9
@backstage/plugin-techdocs-backend@0.7.1
@backstage/plugin-techdocs@0.8.0
@backstage/plugin-user-settings@0.2.9
@backstage/search-common@0.1.1
@backstage/techdocs-common@0.5.0
@backstage/test-utils-core@0.1.1
@backstage/test-utils@0.1.10
@backstage/theme@0.2.6
```


Following that recent fix to this release commit, it appears we need to update to
`@backstage/plugin-scaffolder-backend@0.11.3`
and (maybe) `@backstage/plugin-shortcuts@0.1.3`
?
